src/game/UINationSelectDlg.cpp

```diff
@@ -333,3 +216,104 @@ void CUINationSelectDlg::Render() {
         CN3Base::s_lpD3DDev->SetTextureStageState(0, D3DTSS_ALPHAARG2, dwTexAlpha2);
     }
 }
+
+bool CUINationSelectDlg::ReceiveMessage(CN3UIBase * pSender, DWORD dwMsg) {
+    if (!pSender) {
+        return false;
+    }
+
+    if (!m_bTransitionActive && dwMsg == UIMSG_BUTTON_CLICK) {
+        if (pSender == (CN3UIBase *)m_pBtnCloseKa || pSender == (CN3UIBase *)m_pBtnCloseEl) {
+            CGameProcedure::s_pSocket->Disconnect();
+            CGameProcedure::ProcActiveSet((CGameProcedure *)CGameProcedure::s_pProcLogIn);
+            CGameProcedure::s_pProcLogIn->m_bLogIn = true;
+            return true;
+        } else if (pSender == (CN3UIBase *)m_pBtnNextKa || pSender == (CN3UIBase *)m_pBtnNextEl) {
+            ChangeNation((pSender != (CN3UIBase *)m_pBtnNextKa));
+            ButtonsSetEnable(false);
+            return true;
+        } else if (pSender == (CN3UIBase *)m_pBtnSelectionKa || pSender == (CN3UIBase *)m_pBtnSelectionEl) {
+            m_eCurNation = (pSender == (CN3UIBase *)m_pBtnSelectionKa) ? NATION_KARUS : NATION_ELMORAD;
+            m_pProcNationSelectRef->MsgSendNationSelect(m_eCurNation);
+            return true;
+        }
+    }
+
+    return true;
+}
+
+void CUINationSelectDlg::InitResources() {
+    std::string szBuff;
+    if (m_pTextNoticeEl) {
+        // El Morad is a nation that was formed by King Manes...
+        ::_LoadStringFromResource(IDS_NATIONSELECT_DESCRIPTION_EL, szBuff);
+        m_pTextNoticeEl->SetString(szBuff);
+        m_pTextNoticeEl->SetFont(m_pTextNoticeEl->GetFontName(), m_pTextNoticeEl->GetFontHeight() - 2, FALSE, FALSE);
+    }
+    if (m_pTextNoticeKa) {
+        // Karus is a new nation that was established in Adonis...
+        ::_LoadStringFromResource(IDS_NATIONSELECT_DESCRIPTION_KA, szBuff);
+        m_pTextNoticeKa->SetString(szBuff);
+        m_pTextNoticeKa->SetFont(m_pTextNoticeKa->GetFontName(), m_pTextNoticeKa->GetFontHeight() - 2, FALSE, FALSE);
+    }
+
+    if (m_pBaseEl) {
+        m_pBaseEl->SetVisible(false);
+    }
+    if (m_pBaseKa) {
+        m_pBaseKa->SetVisible(true);
+    }
+    m_bNationKa = true;
+
+    if (m_pSndKa) {
+        CN3Base::s_SndMgr.ReleaseStreamObj(&m_pSndKa);
+    }
+    m_pSndKa = CN3Base::s_SndMgr.CreateStreamObj(ID_SOUND_BGM_KA_BATTLE);
+    if (m_pSndEl) {
+        CN3Base::s_SndMgr.ReleaseStreamObj(&m_pSndEl);
+    }
+    m_pSndEl = CN3Base::s_SndMgr.CreateStreamObj(ID_SOUND_BGM_EL_BATTLE);
+
+    if (m_pSndKa) {
+        m_pSndKa->Looping(true);
+        m_pSndKa->Play(NULL, 0.0f, 3.0f);
+    }
+}
+
+void CUINationSelectDlg::ChangeNation(bool bNationKa) {
+    if (bNationKa) {
+        m_bNationKa = true;
+        if (m_pSndEl) {
+            m_pSndEl->Stop(3.0f);
+        }
+        if (m_pSndKa) {
+            m_pSndKa->Looping(true);
+            m_pSndKa->Play(NULL, 0.0f, 6.0f);
+        }
```

Comment:
From what I've seen, we always set this audio to looping `true`. I suggest moving this into `CUINationSelectDlg::InitResources` for both KA and EL.
```suggestion
        if (m_pSndKa) {
            m_pSndKa->Play(NULL, 0.0f, 6.0f);
        }
```

Reply:
Good point! Please see commit: https://github.com/ko4life-net/ko/pull/125/commits/2fc84fbf2d675725b24d26cc7242fda4218120c2